### ISSUE TITLE
Suggestion: Add HEX to HSL/HSB conversion and inverse

### ISSUE THREAD

**Issue by matteo-de-carli** [luisfontes19/vscode-swissknife#13]
Hi, 
So far I'm loving your extension, it's becoming more and more useful over time for me.

I'm mostly using the color conversion utilities, so some things I'd love to see in this extension would be:
- HEX to HSL conversion
- HSL to HEX conversion
- HEX to HWB conversion
- HWB to HEX conversion

I'm not sure if this is possible, but it would be great if you could add these conversions as well, it'd be even more useful than it already is.

Thanks in advance,
Matteo

**Comment by luisfontes19** [comment]
Hi @matteo-de-carli 

Just to give you an update I'm working on it...

I will only implement a single convertion for HSL and HWB. 

Otherwise I would need to implement all possible combinations like HEX to HSL, RGB to HSL, HWB to HSL, etc. And that would be a bit noisy in the menu section 

You can always invoke multiple scripts to go from the format you have to the format you need. Also you can create user scripts to do conversions directly from the ones you need, just by invoking the native conversion methods. I'll provide you with a working example for that when I close this one :)

**Comment by luisfontes19** [comment]
Hi @matteo-de-carli 

Sorry for the late reply... I've been focusing on other projects... 

Anyway, this is now done and released in 1.8.1

The conversions are to and from Hex (hsl to hex, hex to hsl, etc) otherwise I would end up creating a lot of scripts to cover all possible combinations.

If you want a specific combination you can create your own custom script to achieve this...

Take a look on how to create custom scripts [here](https://github.com/luisfontes19/vscode-swissknife/blob/master/data/custom_scripts.md)

Also, I'll leave here the code to do a custom script to do HSL to HWB, in case it helps:

```js
Object.defineProperty(exports, "__esModule", { value: true });

exports.hslToHwb = async (text) => {

}

const scripts = [
  {
    title: "HSL to HWB",
    detail: "Convert HSL color to HWB",
    cb: (context) => {
      const hslToHwb = async (text) => {
        const res = text.match(/hsl\((\d+),\s*(\d+)%,\s*(\d+)%\)?/)
        if (!res) {
          context.informationRoutine(async () => "Invalid HSL Format")
          return text
        }

        res.shift()
        const [h, s, l] = res.map(Number)


        const hex = context.modules.colors.hslToHex(h, s, l)
        const hwb = context.modules.colors.hexToHwb(hex)

        return Promise.resolve(hwb)
      }
      return context.replaceRoutine(hslToHwb)
    }
  }
]

exports.default = scripts;
```

Also, for reference, the variable context.modules holds a variable for each file in the [lib folder](https://github.com/luisfontes19/vscode-swissknife/tree/master/src/lib) and the methods for each are the same you can find in those files :)
